SalesforceCommerceCloud/pwa-kit
# @W-15621253 Initial Store Locator Implementation

# Description
[W-15621253](https://gus.lightning.force.com/a07EE00001q8J2yYAE)

This is the initial Store Locator implementation, which is without pagination and linking of geolocation web API. The initial implementation includes: 
1. the handling of manual geolocation input
2. default country and postal code and user specified store locator countries
3. the listing of corresponding stores sorted by distance away
4. mobile view
5. a store locator page, on top of the modal, that can be navigated to via URL for SEO

The PR will merge into a store locator base branch, to which subsequent PRs will be merged until the store locator is ready for shipping. Subsequent PRs will link the geolocation web API to the store locator and enable pagination for the store locator.

# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- (change1) Implemented initial store locator page and modal
- (change2) Linked the existing store locator link in the footer to the page
- (change3) Added store locator icon at the header

# How to Test-Drive This PR

- (step1) Ensure all tests pass with `npm run test`
- (step 2) Go to https://runtime-admin-n16.mobify-storefront.com/ and click on the store icon on the top right of the page to open store locator modal
- (step 3) See that the list of stores are sorted by distance away
- (step 4) See that clicking View More opens an accordion to view store hours, if specified
- (step 4) Change location to Country: United States, Postal Code: 94086, and press "Find" to see that new stores are now listed
- (step 5) Close the modal, and open it again to see that the last submitted user input is saved.
- (step 7) Go to https://runtime-admin-n16.mobify-storefront.com/store-locator for the page version of the store locator

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
